### DISCUSSION TITLE
Crop images before uploading

### DISCUSSION THREAD

**Discussion by PegasisForever** [payloadcms/payload#1397] [Feature Requests & Ideas]
It would be very useful to add an option to the upload field which shows users an UI to crop their image to a specific aspect ratio.

**Comment by coxmi** [comment]
Would love to see more fleshed-out media collection settings: 

- Thumbnail grid view
- Drag/drop a folder into the grid view for multiple uploads
- Focal point selection
- CMS cropping
- Config-defined default cropping

Is something like this planned in the roadmap?

**Reply by JosefBredereck** [reply]
> Thumbnail grid view

like this? https://github.com/payloadcms/payload/pull/2849
Or do you mean the overview page?

> Config-defined default cropping

should be present with `fit` + `position` property, but this is just a general direction.

All the other points would be awesome

**Comment by ssyberg** [comment]
👍🏼 bumping / watching

**Comment by STR1234** [comment]
I like the idea. Maybe this could be flashed out with the option to be able to:

- specify a name for the cropped version - keeping the original intact
-  later being able to add additional differently cropped versions with different names to identify them

Could come in handy in use cases where in the frontend you want to e.g. display a profile image but the image you have has multiple people on it and you need to crop out the face of the respective person but still might want to display the original full version in another way. Something like this would keep it pretty clean.

**Reply by olarsson** [reply]
love the idea with later being able to change crop options, affected images would need to be automatically regenerated to match the updated crop settings.

**Comment by emilkarl** [comment]
Lets goo :D

**Comment by olarsson** [comment]
would also love to see this: 

- ability to set individual crop settings for images. might need to rethink the whole way of how media works for this, rather than selecting media directly from an upload collection you might need a middle step where you import a "media cropper" based on a model you create, where the individual cropping settings exist.
- set image quality more freely as well, like appending "?quality=85" to the image url would give return that image, with the crop settings selected, in 85% quality of the original
- a simple quick way of tagging media and filtering by tags as a default. or maybe a folder structure. but basically something to organize them better in the collection.

**Comment by JayBox325** [comment]
I'm using Astro for my frontends so setting quality/cropping is handled in the frontend, but for me the feature I'm eager to have is a Focal Point to be able to correctly configure the cropping around that focal point.
